src/wp-includes/theme.php

```diff
@@ -3770,8 +3771,13 @@ function _wp_customize_loader_settings() {
  */
 function wp_customize_url( $stylesheet = '' ) {
 	$url = admin_url( 'customize.php' );
+
 	if ( $stylesheet ) {
-		$url .= '?theme=' . urlencode( $stylesheet );
+		// Make sure any existing theme argument is removed.
+		$url = remove_query_arg( 'theme', $url );
+
+		// Add the theme argument to the URL.
+		$url = add_query_arg( 'theme', urlencode( $stylesheet ), $url );
```

Comment:
One of the test cases could include a scenario where the `$stylesheet` argument contains a character which gets URL-encoded, just to see that this is working as expected. For example, consider a theme name that has a space or an ampersand in it (even though this would be very unusual).